pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -232,42 +247,36 @@ contract StablePhantomPool is IRateProvider, BaseGeneralPool, ProtocolFeeCache {
         // reference them by token index in the full base tokens plus BPT set (i.e. the tokens the Pool registers). Due
         // to immutable variables requiring an explicit assignment instead of defaulting to an empty value, it is
         // simpler to create a new memory array with the values we want to assign to the immutable state variables.
-        IRateProvider[] memory tokensAndBPTRateProviders = new IRateProvider[](params.tokens.length + 1);
-        for (uint256 i = 0; i < tokensAndBPTRateProviders.length; ++i) {
+        IRateProvider[] memory rateProviders = new IRateProvider[](params.tokens.length + 1);
+        // Do the same with exemptFromYieldProtocolFeeFlags
+        bool[] memory exemptFromYieldFlags = new bool[](rateProviders.length);
+
+        for (uint256 i = 0; i < rateProviders.length; ++i) {
             if (i < bptIndex) {
-                tokensAndBPTRateProviders[i] = params.rateProviders[i];
+                rateProviders[i] = params.rateProviders[i];
+                exemptFromYieldFlags[i] = params.exemptFromYieldProtocolFeeFlags[i];
```

Comment:
```suggestion
                rateProviders[i] = params.rateProviders[i];
                exemptFromYieldFlags[i] = params.exemptFromYieldProtocolFeeFlags[i] && rateProviders[i] != IRateProvider(0) ;
```

I think it's possible to lock funds if you pass a zero rate provider but say it's exempt from yield protocol fees (a not totally unreasonable thing to do).

If you did this then `_getTokenRateRatios` will attempt to compute the rate ratio for an uninitialised price rate cache. We'll then try to calculate `0 / 0` and everything breaks.

Reply:
Good catch! We'll have to test every combination of everything, and every ordering of operations.

Reply:
Maybe even disallow this case? Aren't the exempt tokens normally composed BPT of other pools, which would by definition have a rate function? If it didn't have a rate function, the rate would always be 1 anyway, so there couldn't be a problem, and there'd be no reason to set the exempt flag.

If it gets passed in this way, it seems more likely to be a mistake. If they mixed up the order of the flags - but it accidentally worked - it would just be silently wrong.

Reply:
On second thoughts this wouldn't have resulted in loss of funds due to recovery mode (proving how good of an idea it is).

> If it gets passed in this way, it seems more likely to be a mistake. If they mixed up the order of the flags - but it accidentally worked - it would just be silently wrong.

True, we should err on the side of being more restrictive when it comes to constructor arguments. It's better to reject them than try and "correct" them but fail.